backends/dpdk/dpdkHelpers.cpp

```diff
@@ -768,26 +768,67 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
                         append_parser_name(parser, IR::ParserState::reject)));
             add_instr(new IR::DpdkLabelStatement(end_label));
         } else if (a->method->name == "add_entry") {
+            auto args = a->expr->arguments;
+            auto argSize = args->size();
+            if (argSize != 3) {
+                ::error(ErrorType::ERR_UNEXPECTED, "Unexpected number of arguments for %1%",
+                            a->method->name);
+                return false;
+            }
             auto action = a->expr->arguments->at(0)->expression;
             auto action_name = action->to<IR::StringLiteral>()->value;
             auto param = a->expr->arguments->at(1)->expression;
+            auto timeout_id = a->expr->arguments->at(2)->expression;
+            if (timeout_id->is<IR::Constant>()) {
+                BUG_CHECK(metadataStruct, "Metadata structure missing unexpectedly!");
+                IR::ID tmo(refmap->newName("timeout_id"));
+                auto timeout = new IR::Member(new IR::PathExpression("m"), tmo);
+                metadataStruct->fields.push_back(new IR::StructField(tmo, timeout_id->type));
+                add_instr(new IR::DpdkMovStatement(timeout, timeout_id));
+                timeout_id = timeout;
+            }
             if (param->is<IR::Member>()) {
                 auto argument = param->to<IR::Member>();
-                add_instr(new IR::DpdkLearnStatement(action_name, argument));
+                add_instr(new IR::DpdkLearnStatement(action_name, timeout_id, argument));
```

Comment:
here the argument order is different. not a big deal, but consistency could make things simpler.

Reply:
`add_entry` has 3 parameters in pna.p4. None of the parameters are optional.
Learn instruction has 2 variants. 
```
learn <action_name> <timeout>                     /// when action_params are empty
learn <action_name> <action_params> <timeout>     /// when action params are non-empty
```
<action_param> is optional in the learn instruction, hence kept it at the end.